### PR TITLE
NRPT-556: update search controller to redact issuedTo based on role

### PR DESCRIPTION
**Pull request by mtCarto**
- [NRPT-556](https://bcmines.atlassian.net/browse/NRPT-556)

### CODE REVIEW

File: api/src/controllers/search.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ let qs = require('qs');
 let mongodb = require('../utils/mongodb');
 let moment = require('moment');
 let fuzzySearch = require('../utils/fuzzySearch');
-let constants = require('../utils/constants/misc')
+let constants = require('../utils/constants/misc');
+let AppRoles = constants.ApplicationRoles;
```
Comment (marklise) [inline api/src/controllers/search.js:11]: Unless modifying, technically should be `const`

**Review by BcGovNeal — APPROVED** [review]

**Review by BcGovNeal — CHANGES_REQUESTED** [review]
I test this again and it seems the `issuedTo` names are showing on the Record list when `issuedTo.read` doesn't have `admin:wf` but `read` does.  Record detail is redacting correctly though.

**Comment by mtCarto** [comment]
I'll take a look and correct this

**Comment by mtCarto** [comment]
Should be working as expected now, requires the extra migration to prevent redaction of company names.

**Review by BcGovNeal — APPROVED** [review]
IssuedTo is displaying as expected

**Review by MaxWardle — APPROVED** [review]
Very nice 👏

**Pull request merged**